lib/ddtrace/contrib/mysql2/instrumentation.rb

```diff
@@ -27,36 +32,43 @@ def query(sql, options = {})
               span.set_tag(Datadog::Ext::Integration::TAG_PEER_SERVICE, span.service)
 
               # Set analytics sample rate
-              Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+              if Datadog.configuration[:mysql2][:analytics_enabled]
+                Contrib::Analytics.set_sample_rate(span, Datadog.configuration[:mysql2][:analytics_sample_rate])
+              end
```

Comment:
Not sure if it's "in scope", but have you considered taking this opportunity to move the configuration to be supplied as part of the env, rather than being global state that is accessed directly?

This is similar to how above we use `datadog_pin.tracer` rather than `Datadog.tracer` to get the correct tracer instance.